diff --git a/lib/commands/build/plugin.js b/lib/commands/build/plugin.js
index d23046aca87ba88f5007612aa8811b650d932688..eec6a7e2303b4a7bf4cb34b0c273c4886f8f8359 100644
--- a/lib/commands/build/plugin.js
+++ b/lib/commands/build/plugin.js
@@ -22,6 +22,16 @@ const getNormalizedName = (name) => {
         throw new clipanion_1.UsageError(`Invalid plugin name "${name}" - it should be "yarn-plugin-<something>"`);
     return `@yarnpkg/${parsing[1]}`;
 };
+
+const stripNodeColonPlugin = {
+  name: 'strip-node-colon',
+  setup({ onResolve, onLoad }) {
+    onResolve({ filter: /^node:/ }, args => {
+      return { path: args.path.slice('node:'.length), external: true }
+    })
+  }
+}
+
 // eslint-disable-next-line arca/no-default-export
 class BuildPluginCommand extends clipanion_1.Command {
     constructor() {
@@ -91,7 +101,7 @@ class BuildPluginCommand extends clipanion_1.Command {
                     logLevel: `silent`,
                     format: `iife`,
                     platform: `node`,
-                    plugins: [dynamicLibResolver],
+                    plugins: [dynamicLibResolver, stripNodeColonPlugin],
                     minify: !this.noMinify,
                     sourcemap: this.sourceMap ? `inline` : false,
                     target: `node${semver_1.default.minVersion(package_json_1.default.engines.node).version}`,
